cranelift-codegen/src/binemit/relaxation.rs

```diff
@@ -135,7 +136,11 @@ pub fn relax_branches(
     let jumptables_size = offset - jumptables;
     let rodata = offset;
 
-    // TODO: Once we have constant pools we'll do some processing here to update offset.
+    let constants: Vec<Constant> = func.dfg.constants.iter().map(|(h, _)| h.clone()).collect();
+    for constant_handle in constants {
+        func.dfg.constants.set_offset(constant_handle, offset);
+        offset += func.dfg.constants.get(constant_handle).len() as u32
+    }
```

Comment:
This took a bit of unpacking. We iterate over `func.dfg.constants`, do a `.map` to discard the values and just get the keys, and we have to `.collect()` them because we later want to mutate the values (which hold the values we just discarded), which we look up using the keys.

Would it be possible to simplify this by:
 - give `ConstantPool` an `entries_mut()` function which returns the same thing as `iter()` but hands out the full `&mut ConstantPoolEntry` as values
 - give `ConstantPoolEntry` its own `set_offset` and `len` functions, which are the two things that `relax_branches` here needs to do
?